Local2.0

ogbabydiesal opened the pull request.

montoyamoraga reviewed (approved):
great!

montoyamoraga reviewed (changes requested):
hey this looks good but i think the file
src/data/examples/en/03_Arrays/03_Walk_Over_2dArray.js
should be included in a separate pull request, right?

suhascv commented:
> hey this looks good but i think the file
> src/data/examples/en/03_Arrays/03_Walk_Over_2dArray.js
> should be included in a separate pull request, right?

yeah its for another PR
I'll change that

davepagurek committed:
Mark tutorials as to do

davepagurek committed:
Merge branch 'main' into 2.0

davepagurek committed:
Merge branch 'main' into merge-main

davepagurek committed:
Make one tutorial exist

davepagurek committed:
Merge branch '2.0' into merge-main

ksen0 committed:
Merge pull request #736 from processing/merge-main

Merge main into 2.0

ksen0 committed:
Merge pull request #740 from processing/main

Update 2.0 to match main

ksen0 committed:
Merge pull request #742 from processing/main

Update to match main

lukeplowden committed:
initial commit

lukeplowden committed:
use beta version

lukeplowden committed:
added filter sketch for testing

davepagurek committed:
Put back all working tutorials

davepagurek committed:
Fix bugs and put back a few WebGL tutorials

davepagurek committed:
Merge pull request #752 from processing/feat/fix-more-webgl-tutorials

Fix bugs and put back a few WebGL tutorials

lukeplowden committed:
add example sketch

ksen0 committed:
Merge pull request #753 from processing/main

Updating 2.0 to match main

davepagurek committed:
Bump versions

davepagurek committed:
Update docs for 2.0

davepagurek committed:
Merge pull request #763 from processing/prepare-for-2.0

Prepare for 2.0

davepagurek committed:
Update some more tutorials for 2.0

davepagurek committed:
Merge pull request #764 from processing/fix/tutorials

Update some more tutorials for 2.0

harneetsahi committed:
Style: replaced outlines with borders for improved accessibility

harneetsahi committed:
Style: fix search bar outline

davepagurek committed:
Merge pull request #774 from webermayank/fix-multiple-p5scriptTag

 fix Reference pages with multiple examples load p5 multiple times

ksen0 committed:
Merge pull request #776 from processing/main

Merge main into 2.0 beta

lukeplowden committed:
tutorial finishing touches

lukeplowden committed:
add cross env for windows support to build reference

lukeplowden committed:
finish up tutorial

lukeplowden committed:
Merge remote-tracking branch 'upstream/2.0' into 2.0

lukeplowden committed:
line endings

lukeplowden committed:
line endings

lukeplowden committed:
frontmatter, picture and metadata for first p5-strands tutorial

davepagurek committed:
Fix metadata

ksen0 committed:
Updated beta.6

davepagurek committed:
Revert package lock

davepagurek committed:
Merge pull request #778 from processing/fix/package-lock

Try to fix 2.0 build errors

davepagurek committed:
Migrate examples to 2.0

davepagurek committed:
Merge pull request #779 from processing/fix/examples

Migrate examples to 2.0

lukeplowden committed:
corrections & adjustments

lukeplowden committed:
line break

lukeplowden committed:
Merge branch '2.0' of https://github.com/lukeplowden/p5.js-website into 2.0

ksen0 committed:
Merge pull request #777 from lukeplowden/2.0

p5.strands first tutorial

davepagurek committed:
Merge pull request #781 from processing/fix/sketch-embed-double-render

Remove CDN script tag from embedded sketches

ksen0 committed:
p5.strands tutorial added GLSL link and minor phrasing edits

ksen0 committed:
Phrasing p5.strands intro adjustments

davepagurek committed:
Handle change to deprecation message format

davepagurek committed:
Update types

davepagurek committed:
Merge pull request #787 from processing/fix/deprecation-message

Handle change to deprecation message format

davepagurek committed:
Merge pull request #737 from processing/p5-sound-versions

Only use old p5.sound on 1.x

ksen0 committed:
hader tutorial - CPU phrasing adjusment

ksen0 committed:
Update beta reference based on 2.0 beta.7

ksen0 committed:
Merge pull request #782 from ksen0/2.0-shader-tutorial-edits

p5.strands tutorial added GLSL link and minor phrasing edits

davepagurek committed:
Merge pull request #788 from ksen0/beta7-refs

Update beta reference based on 2.0 beta.7

davepagurek committed:
Merge pull request #792 from processing/fix/p5sound

Fix/p5sound

ffd8 committed:
adding XYscope.js to libs

ksen0 committed:
Docs updated for 2.0 beta.8

ksen0 committed:
Signpost that the beta site is for p5.js 2.0

ksen0 committed:
Updated hero image

ksen0 committed:
Merge pull request #798 from processing/docs-beta8

Added beta8 documentation and beta site signposting

davepagurek committed:
Merge branch 'main' into update-2.0

davepagurek committed:
Update search indices

ksen0 committed:
Merge pull request #803 from processing/update-2.0

Update 2.0

ksen0 committed:
Fix beta npm version script

github-actions[bot] committed:
Update p5.js to v2.0.0

ksen0 committed:
Update 2.0 documentation

ksen0 committed:
Merge pull request #809 from processing/new-20-docs

Update 2.0 documentation

davepagurek committed:
Fully unmount sketches that go out of frame

davepagurek committed:
Merge pull request #812 from processing/feat/perf

Fully unmount sketches that go out of frame

ksen0 committed:
Temporary fix on 2.0 reference assets URLs

ksen0 committed:
Merge pull request #813 from processing/beta-assets-url-patch

Temporary fix on 2.0 reference assets URLs

ksen0 committed:
p5.strands tutorial typo patch

ksen0 committed:
Merge pull request #815 from processing/strands-tut-typo-path

p5.strands tutorial typo patch

ksen0 committed:
Merge branch '2.0' into beta-npm-version

ksen0 committed:
Merge pull request #807 from processing/beta-npm-version

Fix beta npm version script

github-actions[bot] committed:
Update p5.js to v2.0.1

davepagurek committed:
Merge pull request #819 from processing/more-libraries

Add libraries, fix bug with old featured ones not getting cycled

davepagurek committed:
Merge pull request #816 from taylorvance/main

FPS library

shivasankaran18 committed:
fix:removed linter warnings

Signed-off-by: shiva <shiva_05@outlook.com>

webermayank committed:
updated reference regex to not break on beta versions

webermayank committed:
updated regex to remove Unnecessary escape character

clairep94 committed:
fix .lowerCase error

clairep94 committed:
minimal error handling with no return value for errors

clairep94 committed:
add minimal error handling to other functions

clairep94 committed:
update community 10 sketches to be fetched with getCuratedSketches with no args and memoize getCurationSketches

davepagurek committed:
Support rest types in Syntax section of reference

davepagurek committed:
Update ReferenceItem type

clairep94 committed:
remove call to getSketchSize

clairep94 committed:
Add note

clairep94 committed:
refactor getSketch to first check for sketchId. with test console log

clairep94 committed:
destructure return from getSketch for legibility and revert title to come from getSketch

clairep94 committed:
test if getSketchSize is necessary

clairep94 committed:
remove comments about getSketcheSize, should be implemented after OP api is updated

clairep94 committed:
cleanup

ksen0 committed:
Merge branch '2.0' into update-reference-regex-for-2.0

ksen0 committed:
Merge pull request #842 from ksen0/pr-838-cherrypick

PR838 cherrypick into 2.0

ksen0 committed:
Merge branch '2.0' into update-reference-regex-for-2.0

davepagurek committed:
Merge branch '2.0' into rest-types

davepagurek committed:
Merge pull request #836 from processing/rest-types

Support rest types in Syntax section of reference

github-actions[bot] committed:
Update p5.js to v2.0.2

humanbydefinition committed:
Update p5.asciify yaml file name and content

limzykenneth committed:
Merge pull request #844 from limzykenneth/fix-reference-history

Set iframe srcdoc attribute to blank to prevent iframe navigation

davepagurek committed:
Merge pull request #846 from humanbydefinition/2.0

Updated `p5.asciify` add-on library content

perminder-17 committed:
Updating the docs for triangle_strip

davepagurek committed:
Merge pull request #848 from processing/perminder-17-patch-1

Updating the docs for triangle_strip

github-actions[bot] committed:
Update p5.js to v2.0.3

ksen0 committed:
Merge pull request #843 from shivasankaran18/2.0

Removed linter warnings

jishnunambiarr committed:
updated tutorial against 2.0 branch

ksen0 committed:
Merge branch '2.0' into update-reference-regex-for-2.0

ksen0 committed:
Update reference.ts

ksen0 committed:
Merge pull request #826 from webermayank/update-reference-regex-for-2.0

updated reference regex to not break on beta versions (2.0 branch)

davepagurek committed:
Merge pull request #861 from processing/woff2-fix

Fix incorrect link for p5.woff2

perminder-17 committed:
some fixes Co-authored-by: Mamatha1718

perminder-17 committed:
some-directory-fixes

perminder-17 committed:
fixed misspelling

perminder-17 committed:
Using loadImage and removing helper-function

perminder-17 committed:
Merge pull request #825 from Mamatha1718/example

Adding documentation Example on how to do parallel loading with Promi…

jishnunambiarr committed:
Updated tutorial with suggested changes: Lines - 559, 149 and 245

perminder-17 committed:
Merge branch '2.0' into fix-against-2.0

perminder-17 committed:
Merge pull request #859 from jishnunambiarr/fix-against-2.0

Updated custom-shapes-and-smooth-curves.mdx

ksen0 committed:
Update animating-with-media-objects.mdx - remove mention of preload()

Although the tutorial has been updated to use async/await, it still mentions preload() in one spot.

perminder-17 committed:
Merge pull request #876 from processing/pr-875

Update animating-with-media-objects.mdx - remove mention of preload()

NalinDalal committed:
solves docs issue 837 for p5.js 2.x

reshma045 committed:
Fixes #881

Yo7ee committed:
Fixes #882

reshma045 committed:
Update ReferenceLayout.astro

eslteacher902010 committed:
Add aria-labels and accessibility improvements to buttons and navigation components

reshma045 committed:
Fixes #868

reshma045 committed:
updated

davepagurek committed:
Fix escaped characters in inline examples + instance mode examples

davepagurek committed:
Merge pull request #897 from processing/fix/inline-examples-instance

Fix escaped characters in inline examples + instance mode examples

perminder-17 committed:
Merge branch '2.0' into issue868-branch

perminder-17 committed:
Merge pull request #896 from reshma045/issue868-branch

Rename conflicting sphere() functions to avoid shadowing p5.js built-in

perminder-17 committed:
fixing cdn-link for making the p5.Sound library working.

reshma045 committed:
Fixes #887

NalinDalal committed:
updated file name, output format changes, links updated

ksen0 committed:
Merge pull request #898 from perminder-17/2.0

Fixing the CDN-Link for p5.Sound.

NalinDalal committed:
Merge branch '2.0' into fix-on-2.0

tsengwoody committed:
Add screen reader announcement message after CopyCodeButton completes action

tsengwoody committed:
Refactor screen reader announcement message into a reusable hook

ksen0 committed:
Merge pull request #904 from processing/2.0-891

Fixes Screen reader and visual presentation are inconsistent (beta site)

tsengwoody committed:
Add screen reader announcements for Reset, Run, and Stop sketch actions within code blocks

NalinDalal committed:
Merge branch '2.0' into fix-on-2.0

NalinDalal committed:
Merge branch '2.0' into fix-on-2.0

eslteacher902010 committed:
Fix JSX indentation in JumpToLinks

eslteacher902010 committed:
Fix JSX indentation in MainNavLinks toggle

NalinDalal committed:
updated file

tsengwoody committed:
fix(useLiveRegion): clear pending timers to prevent race conditions & leaks

ksen0 committed:
Merge pull request #906 from processing/2.0-905

Add aria-live status messages for button actions to improve screen reader feedback (fixes issue #866 on 2.0 branch)

perminder-17 committed:
Merge branch '2.0' into fix-on-2.0

perminder-17 committed:
Merge pull request #879 from NalinDalal/fix-on-2.0

solves docs issue 837 for p5.js 2.x

ksen0 committed:
Merge pull request #911 from eslteacher902010/aria-labels-2.0

Aria labels 2.0

reshma045 committed:
issue properly fixed

reshma045 committed:
fixed unnecessary inversion of photos

davepagurek committed:
Add Greetings from p5.js 2.0 tutorial

davepagurek committed:
Add new prop to type

lirenjie95 committed:
Fix: Use import for fes.svg instead of copying to public

ksen0 committed:
Merge branch '2.0' into pr-766-20

ksen0 committed:
Merge pull request #921 from processing/pr-766-20

Cherrypick PR 766 to the 2.0 branch

davepagurek committed:
Put p5.strands tutorial in 2.0 category

davepagurek committed:
Shorten alt text

davepagurek committed:
Rearrange sections, link to describe tutorial

davepagurek committed:
Merge pull request #920 from processing/greetings-from-p5

Add Greetings from p5.js 2.0 tutorial

ksen0 committed:
Merge branch '2.0' into pr-902-2.0

ksen0 committed:
Merge pull request #923 from processing/pr-902-2.0

Fix: Improve image visibility in dark mode by inverting images [2.0 branch]

Kakashi786 committed:
Accessibility: Differentiate external footer links with icon and tooltip

github-actions[bot] committed:
Update p5.js to v2.0.4

MissTipo committed:
feat(community): expand curated sketches to include 2025 set

- Updated `getCurationSketches` in `src/api/OpenProcessing.ts` to:
  - Remove hardcoded fetch limit to retrieve all curated sketches
  - Support fetching from both 2024 and 2025 curation IDs
  - Add `curation` property to each sketch for filtering and ordering
- Updated `src/pages/community.astro` to:
  - Use a priority ID list to select and order featured sketches
  - Ensure no duplicate sketches appear in the display
- Added high-resolution images for 2025 curated sketches

Signed-off-by: MissTipo <dorine.a.tipo@gmail.com>

MissTipo committed:
Remove redundant duplicate filter

Signed-off-by: MissTipo <dorine.a.tipo@gmail.com>

davepagurek committed:
Merge pull request #936 from MissTipo/community/sketches

Expand curated sketches to include 2025 set

ksen0 committed:
Updated donate widget with new campaign information

ksen0 committed:
Add back to school fundraiser text

adi-ray committed:
Fix Markdown backtick escaping in Variables and Change tutorial

perminder-17 committed:
Merge pull request #945 from adi-ray/patch-1

Fix Markdown backtick escaping in Variables and Change tutorial [2.0 branch]

ksen0 committed:
Merge branch '2.0' into donate-button

adi-ray committed:
docs: Update reference generation instructions to p5.js 2.0 fork workflow

ksen0 committed:
Merge pull request #943 from ksen0/donate-button

Updated donate widget with new campaign information

github-actions[bot] committed:
Update p5.js to v2.0.5

perminder-17 committed:
Add files via upload

adi-ray committed:
docs: Add npm install step

adi-ray committed:
docs: Clarify branch compatibility

adi-ray committed:
docs: Add terminal output ss

junseok44 committed:
Fix OpenProcessing visualID type mismatch causing rate limiting

- Change visualID type from string to number to match API response
- Update all OpenProcessing functions to accept number parameters
- Add explicit String() conversion in getStaticPaths for URL compatibility
- Remove unnecessary type conversions in sketch lookups
- Enable proper memoization to reduce API calls and prevent 429 errors

ksen0 committed:
Merge pull request #960 from processing/OP-fix-2.0

Fix OpenProcessing visualID type mismatch causing rate limiting [2.0]

ksen0 committed:
Merge branch '2.0' into adding-images-for-splineDocs

ksen0 committed:
Merge pull request #954 from perminder-17/adding-images-for-splineDocs

Adding images for splineDocs

perminder-17 committed:
Merge branch '2.0' into doc-update

ksen0 committed:
Merge pull request #961 from processing/pr-927-2.0

Accessibility: Differentiate external footer links with icon and tooltip

ksen0 committed:
Merge pull request #962 from processing/pr-889-2.0

Fix: Remove Empty Anchor Link in Jump-To Menu on Reference Pages

perminder-17 committed:
Merge branch '2.0' into doc-update

perminder-17 committed:
Merge pull request #948 from adi-ray/doc-update

Update reference generation instructions for p5.js 2.0 fork workflow
